http/typescript/fetch/src/middlewares/middlewareContext.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+import { FetchRequestInfo, FetchRequestInit, FetchResponse } from "../utils/fetchDefinitions";
+import { MiddlewareControl } from "./middlewareControl";
+
+/**
+ * @interface
+ * @property {RequestInfo} request - The request url string or the Request instance
+ * @property {RequestInit} [options] - The options for the request
+ * @property {Response} [response] - The response content
+ *
+ */
+
+export interface MiddlewareContext {
```

Comment:
I strongly prefer the use of a context object. It is a middleware convention in JavaScript. It improves readability for a JS user familiar with a middleware pattern.

The conventional pattern is that the context contains the request, response and all middleware information is required by the handler chain.
The advantage is an extensible parameter list that can be modified by adding more parameters in the future similar to the config object pattern. The users can extend the context object and customize the options on the fly. 

References
https://github.com/koajs/koa/blob/master/docs/api/context.md
https://evertpot.com/generic-middleware/
https://medium.com/@masnun/typescript-with-koa-writing-middlewares-aa1592e4770f

Reply:
Typescript would be the only language to differ, dotnet, java, and go don't have a context object. Ruby doesn't have middleware support yet. I do think this should be part of a larger discussion instead of funneled in this PR.
Would you mind reverting the context aspect on this PR, and opening an issue to discuss that at large instead?